### PR TITLE
modal is responsive

### PR DESCRIPTION
**Pull request by AvnerGuyAvrahamCom**

### CODE REVIEW

File: src/components/common/ModalComponent.tsx [hunk 1]
```diff
@@ -28,21 +28,26 @@ const ModalComponent: FC<IModalProps> = ({
...
             top={0}
             bgColor="blackAlpha.800">
-            <Box w="70vw">
+            <Box w={{ md: "70vw", base: "100%" }}>
               <Flex minHeight="fit-content" direction="row-reverse">
                 <IconButton
+                  display={{ base: "none", md: "block" }}
                   marginRight="-10"
                   borderRadius="50%"
                   onClick={onToggle}
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:37]: disable border radius on small screens

File: src/components/common/ModalComponent.tsx [hunk 2]
```diff
@@ -28,21 +28,26 @@ const ModalComponent: FC<IModalProps> = ({
...
                   aria-label="close modal"
                   icon={<CloseIcon />}
                 />
               </Flex>
-              <Box bgColor="white" borderRadius="sm" h="80vh" p="5">
+
+              <Box bgColor="white" borderRadius="sm" h={{ md: "80vh", base: "100vh" }} p="5">
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:43]: remove border radius on small screens

File: src/components/common/ModalComponent.tsx [hunk 3]
```diff
@@ -28,21 +28,26 @@ const ModalComponent: FC<IModalProps> = ({
...
                 <Flex h="100%" direction={"column"}>
                   <Box h="10%">{ModalHeader}</Box>
                   <Box h="80%">{ModalBody}</Box>
                   <Box h="10%">{ModalFooter}</Box>
+                  <Button display={{ base: "block", md: "none" }} onClick={() => onToggle()}>
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:48]: use `useBreakpointValue` for not rendering the button when the computer is md and up

File: src/components/common/ModalComponent.tsx [hunk 4]
```diff
@@ -28,21 +28,26 @@ const ModalComponent: FC<IModalProps> = ({
...
                   borderRadius="50%"
                   onClick={onToggle}
                   aria-label="close modal"
                   icon={<CloseIcon />}
                 />
               </Flex>
-              <Box bgColor="white" borderRadius="sm" h="80vh" p="5">
+
+              <Box bgColor="white" borderRadius="sm" h={{ md: "80vh", base: "100vh" }} p="5">
                 <Flex h="100%" direction={"column"}>
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:44]: remove all curly braces from strings that are like this `{"column"}`

File: src/components/common/ModalComponent.tsx [hunk 5]
```diff
@@ -28,21 +28,26 @@ const ModalComponent: FC<IModalProps> = ({
             position="fixed"
             top={0}
             bgColor="blackAlpha.800">
-            <Box w="70vw">
+            <Box w={{ md: "70vw", base: "100%" }}>
               <Flex minHeight="fit-content" direction="row-reverse">
                 <IconButton
+                  display={{ base: "none", md: "block" }}
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:34]: don't render this button, just like the close modal button use the `useBreakpointValue` hook, (use the same one)

File: src/components/common/ModalComponent.tsx [hunk 6]
```diff
@@ -28,21 +28,26 @@ const ModalComponent: FC<IModalProps> = ({
...
                   icon={<CloseIcon />}
                 />
               </Flex>
-              <Box bgColor="white" borderRadius="sm" h="80vh" p="5">
+
+              <Box bgColor="white" borderRadius="sm" h={{ md: "80vh", base: "100vh" }} p="5">
                 <Flex h="100%" direction={"column"}>
                   <Box h="10%">{ModalHeader}</Box>
                   <Box h="80%">{ModalBody}</Box>
                   <Box h="10%">{ModalFooter}</Box>
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:47]: make a `handleCloseModal` function

File: src/components/common/ModalComponent.tsx [hunk 7]
```diff
@@ -28,21 +32,28 @@ const ModalComponent: FC<IModalProps> = ({
...
-                  onClick={onToggle}
-                  aria-label="close modal"
-                  icon={<CloseIcon />}
-                />
-              </Flex>
-              <Box bgColor="white" borderRadius="sm" h="80vh" p="5">
-                <Flex h="100%" direction={"column"}>
+            <Box w={{ md: "70vw", base: "100%" }}>
+              {!display && (
+                <Flex minHeight="fit-content" direction="row-reverse">
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:37]: why do you have fit-content?

Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:37]: change `direction` to `justifyContent="flex-end`

File: src/components/common/ModalComponent.tsx [hunk 8]
```diff
@@ -1,21 +1,25 @@
...
   ModalBody,
   ModalFooter,
   isOpen,
-  onToggle,
+  onClose,
 }) => {
+  const display = useBreakpointValue({ base: true, md: false });
+  const handleCloseModal = () => {
+    onClose();
+  };
```
Comment (EkkoKo) [inline src/components/common/ModalComponent.tsx:22]: add a space here

File: src/components/common/ModalComponent.tsx (hunk 7 above)
Comment (AvnerGuyAvrahamCom) [reply src/components/common/ModalComponent.tsx]: done

Comment (AvnerGuyAvrahamCom) [reply src/components/common/ModalComponent.tsx]: done

File: src/components/common/ModalComponent.tsx (hunk 8 above)
Comment (AvnerGuyAvrahamCom) [reply src/components/common/ModalComponent.tsx:23]: done

File: src/components/common/ModalComponent.tsx (hunk 2 above)
Comment (AvnerGuyAvrahamCom) [reply src/components/common/ModalComponent.tsx]: teat

**Pull request merged**